llvm/lib/Support/APInt.cpp

```diff
@@ -74,11 +66,14 @@ inline static unsigned getDigit(char cdigit, uint8_t radix) {
 
 
 void APInt::initSlowCase(uint64_t val, bool isSigned) {
-  U.pVal = getClearedMemory(getNumWords());
-  U.pVal[0] = val;
-  if (isSigned && int64_t(val) < 0)
-    for (unsigned i = 1; i < getNumWords(); ++i)
-      U.pVal[i] = WORDTYPE_MAX;
+  if (isSigned && int64_t(val) < 0) {
+    U.pVal = new WordType[getNumWords()];
+    U.pVal[0] = val;
+    memset(&U.pVal[1], 0xFF, sizeof(WordType) * (getNumWords() - 1));
+  } else {
+    U.pVal = new WordType[getNumWords()]();
+    U.pVal[0] = val;
+  }
   clearUnusedBits();
```

Comment:
Is clearUnusedBits needed on the zero initialized path?

Reply:
Yes, it's needed to clear the higher-order bits that aren't part of the APInt's precision.

Reply:
Wouldn't they already be 0 from the default initialization?

Reply:
Yes, I believe `clearUnusedBits()` can be moved into the non-default-initialization path.